src/FsAutoComplete.Core/RecordStubGenerator.fs

```diff
@@ -185,310 +186,27 @@ let formatRecord (insertionPos: RecordStubsInsertionParams) (fieldDefaultValue:
-                List.tryPick walkExpr [synExpr1; synExpr2]
-            | SynExpr.NamedIndexedPropertySet(_longIdent, synExpr1, synExpr2, _range) ->
-                List.tryPick walkExpr [synExpr1; synExpr2]
-
-            | SynExpr.DotNamedIndexedPropertySet(synExpr1, _longIdent, synExpr2, synExpr3, _range) ->
-                List.tryPick walkExpr [synExpr1; synExpr2; synExpr3]
-
-            | SynExpr.TypeTest(synExpr, _synType, _range)
-            | SynExpr.Upcast(synExpr, _synType, _range)
-            | SynExpr.Downcast(synExpr, _synType, _range) ->
-                walkExpr synExpr
-            | SynExpr.InferredUpcast(synExpr, _range)
-            | SynExpr.InferredDowncast(synExpr, _range) ->
-                walkExpr synExpr
-            | SynExpr.AddressOf(_, synExpr, _range, _range2) ->
-                walkExpr synExpr
-            | SynExpr.TraitCall(_synTyparList, _synMemberSig, synExpr, _range) ->
-                walkExpr synExpr
-
-            | SynExpr.Null(_range)
-            | SynExpr.ImplicitZero(_range) ->
-                None
-
-            | SynExpr.YieldOrReturn(_, synExpr, _range)
-            | SynExpr.YieldOrReturnFrom(_, synExpr, _range)
-            | SynExpr.DoBang(synExpr, _range) ->
-                walkExpr synExpr
-
-            | SynExpr.LetOrUseBang(rhs = synExpr1; andBangs = ands; body = synExpr2) ->
-                [ synExpr1
-                  yield! ands |> List.map (fun (SynExprAndBang(body = body)) -> body)
-                  synExpr2 ] |> List.tryPick walkExpr
-
-            | SynExpr.LibraryOnlyILAssembly _
-            | SynExpr.LibraryOnlyStaticOptimization _
-            | SynExpr.LibraryOnlyUnionCaseFieldGet _
-            | SynExpr.LibraryOnlyUnionCaseFieldSet _ ->
-                None
-            | SynExpr.ArbitraryAfterError(_debugStr, _range) ->
-                None
-
-            | SynExpr.FromParseError(synExpr, _range)
-            | SynExpr.DiscardAfterMissingQualificationAfterDot(synExpr, _range) ->
-                walkExpr synExpr
-
-            | _ -> None
-        )
-
-    and walkRecordField (SynExprRecordField(expr = synExprOpt)) =
-        match synExprOpt with
-        | Some synExpr when Range.rangeContainsPos synExpr.Range pos -> walkExpr synExpr
-        | _ -> None
-
-    and walkSynInterfaceImpl (SynInterfaceImpl(bindings = synBindings): SynInterfaceImpl) =
-        List.tryPick walkBinding synBindings
-
-    match parsedInput with
-    | ParsedInput.SigFile _input -> None
-    | ParsedInput.ImplFile input -> walkImplFileInput input
+let walkAndFindRecordBinding (pos, input) =
+  let walker = {
+      new SyntaxVisitorBase<RecordExpr>() with
+        member x.VisitExpr(path: SyntaxVisitorPath, traverseSynExpr: (SynExpr -> _ option), defaultTraverse: (SynExpr -> _ option), synExpr: SynExpr) =
+          match synExpr with
+          | SynExpr.Record (recordFields = recordFields; copyInfo = copyInfo) ->
```

Comment:
this is a lot less code than the other version and expresses the majority of the same intent. any addition we want to make to this will be _much_ easier in the standard walker.